README.md

```diff
@@ -50,54 +59,78 @@ However you can pass in some config options, the defaults are:
   -- Normal mode mapping left hand side
   line_mapping = "gcc",
   -- Visual/Operator mapping left hand side
-  operator_mapping = "gc"
+  operator_mapping = "gc",
+  -- Hook function to call before commenting takes place
+  hook = nil
 }
 ```
 
-**Ignore Empty Lines**
+#### Ignore Empty Lines
+
 ```lua
 require('nvim_comment').setup({comment_empty = false})
 ```
 
-**Disable mappings**
+#### Disable mappings
+
 ```lua
 require('nvim_comment').setup({create_mappings = false})
 ```
 
-**Custom mappings**
+#### Custom mappings
+
 ```lua
 require('nvim_comment').setup({line_mapping = "<leader>cl", operator_mapping = "<leader>c"})
 ```
 
-**Disable marker padding**
+#### Disable marker padding
+
 ```lua
 require('nvim_comment').setup({marker_padding = false})
 ```
 
-**Changing/Setting commentstring**
+#### Hook function called before reading `commentstring`
 
-If you want to override the comment markers or add a new filetype just set the **commentstring** options:
+You can run arbitrary function which will be called before plugin reads value of
+`commentstring`. This can be used to integrate with
+[JoosepAlviste/nvim-ts-context-commentstring](https://github.com/JoosepAlviste/nvim-ts-context-commentstring):
+
+```lua
+require('nvim_comment').setup({
+  hook = function()
+    if vim.api.nvim_buf_get_option(0, "filetype") == "vue" then
+      require("ts_context_commentstring.internal").update_commentstring()
+    end
+  end
+})
+```
+
+#### Changing/Setting commentstring
+
+If you want to override the comment markers or add a new filetype just set the
+**commentstring** options:
 
 ```lua
 -- Assumes this is being run in the context of the filetype...
 vim.api.nvim_buf_set_option(0, "commentstring", "# %s")
 ```
 
-# Installation
+## Installation
```

Comment:
happy with current heading level

Reply:
ignore this then based on your TOC comment

Reply:
Made it a list, not sure if it's good enough now. Perhaps doesn't even need to be a list. Detail though.